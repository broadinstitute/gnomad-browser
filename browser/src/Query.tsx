import React, { Component } from 'react'

import Delayed from './Delayed'
import StatusMessage from './StatusMessage'

const areVariablesEqual = (variables: any, otherVariables: any) => {
  const keys = Object.keys(variables)
  const otherKeys = Object.keys(otherVariables)
  if (keys.length !== otherKeys.length) {
    return false
  }
  return keys.every((key) => variables[key] === otherVariables[key])
}

const cancelable = (promise: any) => {
  let isCanceled = false
  const wrapper = new Promise((resolve: any, reject: any) => {
    promise.then(
      (value: any) => {
        if (!isCanceled) {
          resolve(value)
        }
      },
      (error: any) => {
        if (!isCanceled) {
          reject(error)
        }
      }
    )
  })

  return {
    cancel: () => {
      isCanceled = true
    },
    promise: wrapper,
  }
}

type BaseQueryState = any

type BaseQueryProps = {
  operationName: string | null
  query: string
  url: string
  variables?: any
  children: (state: BaseQueryState) => JSX.Element
}

export class BaseQuery extends Component<BaseQueryProps, BaseQueryState> {
  static defaultProps = {
    url: '/api/',
    operationName: null,
    variables: {},
  }

  currentRequest: any

  mounted: any

  state = {
    data: null,
    error: null,
    graphQLErrors: null,
    loading: true,
  }

  componentDidMount() {
    this.mounted = true
    this.loadData()
  }

  componentDidUpdate(prevProps: BaseQueryProps) {
    const { query, variables } = this.props
    if (query !== prevProps.query || !areVariablesEqual(variables, prevProps.variables)) {
      this.loadData()
    }
  }

  componentWillUnmount() {
    this.mounted = false
  }

  loadData() {
    const { operationName, query, url, variables } = this.props

    this.setState({
      loading: true,
      error: null,
      graphQLErrors: null,
    })

    if (this.currentRequest) {
      this.currentRequest.cancel()
    }

    this.currentRequest = cancelable(
      fetch(url, {
        body: JSON.stringify({
          operationName,
          query,
          variables,
        }),
        method: 'POST',
        headers: {
          'Content-Type': 'application/json',
        },
      }).then((response) => response.json())
    )
    this.currentRequest.promise.then(
      (response: any) => {
        if (!this.mounted) {
          return
        }
        this.setState({
          data: response.data,
          error: null,
          graphQLErrors: response.errors,
          loading: false,
        })
      },
      (error: any) => {
        if (!this.mounted) {
          return
        }
        this.setState({
          data: null,
          error,
          graphQLErrors: null,
          loading: false,
        })
      }
    )
  }

  render() {
    const { children } = this.props
    return children(this.state)
  }
}

type OwnQueryProps = {
  children: (...args: any[]) => any
  errorMessage?: string
  loadingMessage?: string
  loadingPlaceholderHeight?: number
  operationName: string | null
  query: string
  success?: (...args: any[]) => any
  url: string
  variables?: any
}

// @ts-expect-error TS(2456) FIXME: Type alias 'QueryProps' circularly references itse... Remove this comment to see the full error message
type QueryProps = OwnQueryProps & typeof Query.defaultProps

// @ts-expect-error TS(7022) FIXME: 'Query' implicitly has type 'any' because it does ... Remove this comment to see the full error message
const Query = ({
  children,
  errorMessage,
  loadingMessage,
  loadingPlaceholderHeight,
  operationName,
  query,
  success,
  url,
  variables,
}: QueryProps) => {
  return (
    <BaseQuery operationName={operationName} query={query} url={url} variables={variables}>
      {({ data, error, graphQLErrors, loading }: any) => {
        if (loading) {
          return (
            <div style={{ height: loadingPlaceholderHeight || 'auto' }}>
              <Delayed>
                <StatusMessage>{loadingMessage}</StatusMessage>
              </Delayed>
            </div>
          )
        }

        if (error) {
          return <StatusMessage>{errorMessage}</StatusMessage>
        }

        if (!data || !success(data)) {
          return (
            <StatusMessage>
              {graphQLErrors && graphQLErrors.length
                ? Array.from(new Set(graphQLErrors.map((e: any) => e.message))).join(', ')
                : errorMessage}
            </StatusMessage>
          )
        }

        return children({ data })
      }}
    </BaseQuery>
  )
}

Query.defaultProps = {
  errorMessage: 'Error',
  loadingMessage: 'Loading',
  loadingPlaceholderHeight: undefined,
  success: () => true,
  url: '/api/',
  variables: {},
  operationName: null,
}

export default Query
