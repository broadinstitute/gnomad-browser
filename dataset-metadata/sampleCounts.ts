const exacSampleCounts = require('./datasets/exac/sampleCounts')
const {
  subsets: gnomadV2SubsetSampleCounts,
  ...gnomadV2SampleCounts
} = require('./datasets/gnomad-v2/sampleCounts')
const {
  subsets: gnomadV3SubsetSampleCounts,
  ...gnomadV3SampleCounts
} = require('./datasets/gnomad-v3/sampleCounts')
const {
  subsets: gnomadSvV2SubsetSampleCounts,
  ...gnomadSvV2SampleCounts
} = require('./datasets/gnomad-sv-v2/sampleCounts')

const sampleCounts = [
  { exac: exacSampleCounts },
  { gnomad_r2_1: gnomadV2SampleCounts },
  ...Object.keys(gnomadV2SubsetSampleCounts).map((subset) => ({
    [`gnomad_r2_1_${subset}`]: gnomadV2SubsetSampleCounts[subset],
  })),
  { gnomad_r3: gnomadV3SampleCounts },
  ...Object.keys(gnomadV3SubsetSampleCounts).map((subset) => ({
    [`gnomad_r3_${subset}`]: gnomadV3SubsetSampleCounts[subset],
  })),
  { gnomad_sv_r2_1: gnomadSvV2SampleCounts },
  ...Object.keys(gnomadSvV2SubsetSampleCounts).map((subset) => ({
    [`gnomad_sv_r2_1_${subset}`]: gnomadSvV2SubsetSampleCounts[subset],
  })),
  { gnomad_sv_r3: { total: 63046 } },
].reduce(Object.assign, {})

export default sampleCounts
