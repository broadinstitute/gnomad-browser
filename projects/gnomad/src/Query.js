import gqlFetch from 'graphql-fetch'
import PropTypes from 'prop-types'
import { Component } from 'react'

const areVariablesEqual = (variables, otherVariables) => {
  const keys = Object.keys(variables)
  const otherKeys = Object.keys(otherVariables)
  if (keys.length !== otherKeys.length) {
    return false
  }
  return keys.every(key => variables[key] === otherVariables[key])
}

const cancelable = promise => {
  let isCanceled = false
  const wrapper = new Promise((resolve, reject) => {
    promise.then(
      value => {
        if (!isCanceled) {
          resolve(value)
        }
      },
      error => {
        if (!isCanceled) {
          reject(error)
        }
      }
    )
  })

  return {
    cancel: () => {
      isCanceled = true
    },
    promise: wrapper,
  }
}

export class Query extends Component {
  static propTypes = {
    children: PropTypes.func.isRequired,
    query: PropTypes.string.isRequired,
    variables: PropTypes.object,
  }

  static defaultProps = {
    variables: {},
  }

  state = {
    data: null,
    error: null,
    loading: true,
  }

  componentDidMount() {
    this.mounted = true
    this.loadData()
  }

  componentDidUpdate(prevProps) {
    if (
      this.props.query !== prevProps.query ||
      !areVariablesEqual(this.props.variables, prevProps.variables)
    ) {
      this.loadData()
    }
  }

  componentWillUnmount() {
    this.mounted = false
  }

  loadData() {
    this.setState({
      loading: true,
      error: null,
    })

    if (this.currentRequest) {
      this.currentRequest.cancel()
    }

    this.currentRequest = cancelable(
      gqlFetch(process.env.GNOMAD_API_URL)(this.props.query, this.props.variables)
    )
    this.currentRequest.promise.then(
      response => {
        if (!this.mounted) {
          return
        }
        this.setState({
          data: response.data,
          error: null,
          loading: false,
        })
      },
      error => {
        if (!this.mounted) {
          return
        }
        this.setState({
          data: null,
          error,
          loading: false,
        })
      }
    )
  }

  render() {
    return this.props.children(this.state)
  }
}
