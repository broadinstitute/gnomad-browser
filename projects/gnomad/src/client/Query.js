import gqlFetch from 'graphql-fetch'
import PropTypes from 'prop-types'
import { Component } from 'react'

const areVariablesEqual = (variables, otherVariables) => {
  const keys = Object.keys(variables)
  const otherKeys = Object.keys(otherVariables)
  if (keys.length !== otherKeys.length) {
    return false
  }
  return keys.every(key => variables[key] === otherVariables[key])
}

const cancelable = promise => {
  let isCanceled = false
  const wrapper = new Promise((resolve, reject) => {
    promise.then(
      value => {
        if (!isCanceled) {
          resolve(value)
        }
      },
      error => {
        if (!isCanceled) {
          reject(error)
        }
      }
    )
  })

  return {
    cancel: () => {
      isCanceled = true
    },
    promise: wrapper,
  }
}

class Query extends Component {
  static propTypes = {
    children: PropTypes.func.isRequired,
    query: PropTypes.string.isRequired,
    url: PropTypes.string,
    variables: PropTypes.object, // eslint-disable-line react/forbid-prop-types
  }

  static defaultProps = {
    url: '/api/',
    variables: {},
  }

  state = {
    data: null,
    error: null,
    graphQLErrors: null,
    loading: true,
  }

  componentDidMount() {
    this.mounted = true
    this.loadData()
  }

  componentDidUpdate(prevProps) {
    const { query, variables } = this.props
    if (query !== prevProps.query || !areVariablesEqual(variables, prevProps.variables)) {
      this.loadData()
    }
  }

  componentWillUnmount() {
    this.mounted = false
  }

  loadData() {
    const { query, url, variables } = this.props

    this.setState({
      loading: true,
      error: null,
      graphQLErrors: null,
    })

    if (this.currentRequest) {
      this.currentRequest.cancel()
    }

    this.currentRequest = cancelable(gqlFetch(url)(query, variables))
    this.currentRequest.promise.then(
      response => {
        if (!this.mounted) {
          return
        }
        this.setState({
          data: response.data,
          error: null,
          graphQLErrors: response.errors,
          loading: false,
        })
      },
      error => {
        if (!this.mounted) {
          return
        }
        this.setState({
          data: null,
          error,
          graphQLErrors: null,
          loading: false,
        })
      }
    )
  }

  render() {
    const { children } = this.props
    return children(this.state)
  }
}

export default Query
