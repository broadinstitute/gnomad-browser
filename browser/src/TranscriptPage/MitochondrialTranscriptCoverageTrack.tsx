import React from 'react'

import {
  DatasetId,
  labelForDataset,
  referenceGenome,
  hasMitochondrialGenomeCoverage,
} from '@gnomad/dataset-metadata/metadata'
import CoverageTrack from '../CoverageTrack'
import Query from '../Query'
import StatusMessage from '../StatusMessage'

const operationName = 'MitochondrialCoverageInTranscript'
const query = `
query ${operationName}($transcriptId: String!, $datasetId: DatasetId!, $referenceGenome: ReferenceGenomeId!) {
  transcript(transcript_id: $transcriptId, reference_genome: $referenceGenome) {
    mitochondrial_coverage(dataset: $datasetId) {
      pos
      mean
      median
      over_100
      over_1000
    }
  }
}
`

type Props = {
  datasetId: DatasetId
  transcriptId: string
}

const MitochondrialTranscriptCoverageTrack = ({ datasetId, transcriptId }: Props) => {
  if (!hasMitochondrialGenomeCoverage(datasetId)) {
    return (
      <StatusMessage>
        Mitochondrial genome coverage is not available in {labelForDataset(datasetId)}
      </StatusMessage>
    )
  }

  return (
    <Query
      operationName={operationName}
      query={query}
      variables={{ transcriptId, datasetId, referenceGenome: referenceGenome(datasetId) }}
      loadingMessage="Loading coverage"
      loadingPlaceholderHeight={220}
      errorMessage="Unable to load coverage"
      success={(data: any) => data.transcript && data.transcript.mitochondrial_coverage}
    >
      {({ data }: any) => {
        const coverage = [
          {
            color: 'rgb(115, 171, 61)',
            buckets: data.transcript.mitochondrial_coverage,
            name: 'mitochondrial genome',
            opacity: 0.7,
          },
        ]

        return (
          // @ts-expect-error TS(2769) FIXME: No overload matches this call.
          <CoverageTrack
            coverageOverThresholds={[100, 1000]}
            datasets={coverage}
            filenameForExport={() => `${transcriptId}_coverage`}
            height={190}
            maxCoverage={3000}
            datasetId={datasetId}
          />
        )
      }}
    </Query>
  )
}

export default MitochondrialTranscriptCoverageTrack
