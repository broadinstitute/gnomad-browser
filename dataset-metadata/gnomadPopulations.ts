import { textOrMissingTextWarning } from '../browser/src/missingContent'

export const GNOMAD_POPULATION_NAMES = {
  afr: 'African/African American',
  ami: 'Amish',
  amr: 'Admixed American',
  asj: 'Ashkenazi Jewish',
  eas: 'East Asian',
  mid: 'Middle Eastern',
  eur: 'European',
  nfe: 'European (non-Finnish)',
  fin: 'European (Finnish)',
  oth: 'Remaining individuals',
  sas: 'South Asian',

  // EAS subpopulations
  eas_jpn: 'Japanese',
  eas_kor: 'Korean',
  eas_oea: 'Other East Asian',

  // NFE subpopulations
  nfe_bgr: 'Bulgarian',
  nfe_est: 'Estonian',
  nfe_nwe: 'North-western European',
  nfe_onf: 'Other non-Finnish European',
  nfe_seu: 'Southern European',
  nfe_swe: 'Swedish',
} as const

export type PopulationId = keyof typeof GNOMAD_POPULATION_NAMES

export const populationName = (populationId: string) =>
  textOrMissingTextWarning('genetic ancestry group name', GNOMAD_POPULATION_NAMES, populationId)
