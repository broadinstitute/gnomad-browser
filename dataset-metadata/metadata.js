import { datasetLabels } from '../browser/src/datasets'

const fullDatasetIds = Object.getOwnPropertyNames(datasetLabels).filter(
  (datasetId) => datasetId === 'exac' || datasetId.match(/_r\d+(_\d+)*$/)
)

export const isSubset = (datasetId) => !fullDatasetIds.includes(datasetId)
