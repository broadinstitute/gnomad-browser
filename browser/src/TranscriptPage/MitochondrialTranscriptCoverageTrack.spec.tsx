import React from 'react'
import renderer from 'react-test-renderer'
import { createRenderer } from 'react-test-renderer/shallow'

import { jest, describe, expect, test } from '@jest/globals'
import { mockQueries } from '../../../tests/__helpers__/queries'
import Query, { BaseQuery } from '../Query'

import MitochondrialTranscriptCoverageTrack from './MitochondrialTranscriptCoverageTrack'

import { DatasetId, allDatasetIds } from '@gnomad/dataset-metadata/metadata'

jest.mock('../Query', () => {
  const originalModule = jest.requireActual('../Query')

  return {
    __esModule: true,
    ...(originalModule as object),
    default: jest.fn(),
    BaseQuery: jest.fn(),
  }
})

const { resetMockApiCalls, resetMockApiResponses, simulateApiResponse, setMockApiResponses } =
  mockQueries()

beforeEach(() => {
  Query.mockImplementation(
    jest.fn(({ query, children, operationName, variables }) =>
      simulateApiResponse('Query', query, children, operationName, variables)
    )
  )
  ;(BaseQuery as any).mockImplementation(
    jest.fn(({ query, children, operationName, variables }) =>
      simulateApiResponse('BaseQuery', query, children, operationName, variables)
    )
  )
})

afterEach(() => {
  resetMockApiCalls()
  resetMockApiResponses()
})

const datasetsWithoutCoverage: DatasetId[] = [
  'exac',
  'gnomad_r2_1',
  'gnomad_r2_1_controls',
  'gnomad_r2_1_non_neuro',
  'gnomad_r2_1_non_cancer',
  'gnomad_r2_1_non_topmed',
]
const datasetsWithCoverage = allDatasetIds.filter(
  (datasetId) => !datasetsWithoutCoverage.includes(datasetId)
)

describe.each(datasetsWithoutCoverage)(
  'MitochondrialTranscriptCoverageTrack with dataset %s',
  (datasetId) => {
    test('has no unexpected changes', () => {
      const tree = renderer.create(
        <MitochondrialTranscriptCoverageTrack
          datasetId={datasetId}
          transcriptId="dummy-transcript"
        />
      )
      expect(tree).toMatchSnapshot()
    })
  }
)

describe.each(datasetsWithCoverage)(
  'MitochondrialTranscriptCoverageTrack with dataset %s',
  (datasetId) => {
    test('queries with proper parameters', () => {
      setMockApiResponses({
        MitochondrialCoverageInRegion: () => ({}),
      })

      const shallowRenderer = createRenderer()
      shallowRenderer.render(
        <MitochondrialTranscriptCoverageTrack
          datasetId={datasetId}
          transcriptId="dummy-transcript"
        />
      )

      expect(shallowRenderer.getRenderOutput()).toMatchSnapshot()
    })
  }
)
