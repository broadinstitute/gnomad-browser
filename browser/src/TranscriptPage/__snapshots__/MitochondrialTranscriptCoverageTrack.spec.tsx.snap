// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MitochondrialTranscriptCoverageTrack with dataset exac has no unexpected changes 1`] = `
<div
  className="StatusMessage-sc-neg7mh-0 hJFlwG"
>
  Mitochondrial genome coverage is not available in 
  ExAC v1.0
</div>
`;

exports[`MitochondrialTranscriptCoverageTrack with dataset gnomad_r2_1 has no unexpected changes 1`] = `
<div
  className="StatusMessage-sc-neg7mh-0 hJFlwG"
>
  Mitochondrial genome coverage is not available in 
  gnomAD v2.1.1
</div>
`;

exports[`MitochondrialTranscriptCoverageTrack with dataset gnomad_r2_1_controls has no unexpected changes 1`] = `
<div
  className="StatusMessage-sc-neg7mh-0 hJFlwG"
>
  Mitochondrial genome coverage is not available in 
  gnomAD v2.1.1 (controls)
</div>
`;

exports[`MitochondrialTranscriptCoverageTrack with dataset gnomad_r2_1_non_cancer has no unexpected changes 1`] = `
<div
  className="StatusMessage-sc-neg7mh-0 hJFlwG"
>
  Mitochondrial genome coverage is not available in 
  gnomAD v2.1.1 (non-cancer)
</div>
`;

exports[`MitochondrialTranscriptCoverageTrack with dataset gnomad_r2_1_non_neuro has no unexpected changes 1`] = `
<div
  className="StatusMessage-sc-neg7mh-0 hJFlwG"
>
  Mitochondrial genome coverage is not available in 
  gnomAD v2.1.1 (non-neuro)
</div>
`;

exports[`MitochondrialTranscriptCoverageTrack with dataset gnomad_r2_1_non_topmed has no unexpected changes 1`] = `
<div
  className="StatusMessage-sc-neg7mh-0 hJFlwG"
>
  Mitochondrial genome coverage is not available in 
  gnomAD v2.1.1 (non-TOPMed)
</div>
`;

exports[`MitochondrialTranscriptCoverageTrack with dataset gnomad_r3 queries with proper parameters 1`] = `
<mockConstructor
  errorMessage="Unable to load coverage"
  loadingMessage="Loading coverage"
  loadingPlaceholderHeight={220}
  operationName="MitochondrialCoverageInTranscript"
  query="
query MitochondrialCoverageInTranscript($transcriptId: String!, $datasetId: DatasetId!, $referenceGenome: ReferenceGenomeId!) {
  transcript(transcript_id: $transcriptId, reference_genome: $referenceGenome) {
    mitochondrial_coverage(dataset: $datasetId) {
      pos
      mean
      median
      over_100
      over_1000
    }
  }
}
"
  success={[Function]}
  variables={
    {
      "datasetId": "gnomad_r3",
      "referenceGenome": "GRCh38",
      "transcriptId": "dummy-transcript",
    }
  }
>
  [Function]
</mockConstructor>
`;

exports[`MitochondrialTranscriptCoverageTrack with dataset gnomad_r3_controls_and_biobanks queries with proper parameters 1`] = `
<mockConstructor
  errorMessage="Unable to load coverage"
  loadingMessage="Loading coverage"
  loadingPlaceholderHeight={220}
  operationName="MitochondrialCoverageInTranscript"
  query="
query MitochondrialCoverageInTranscript($transcriptId: String!, $datasetId: DatasetId!, $referenceGenome: ReferenceGenomeId!) {
  transcript(transcript_id: $transcriptId, reference_genome: $referenceGenome) {
    mitochondrial_coverage(dataset: $datasetId) {
      pos
      mean
      median
      over_100
      over_1000
    }
  }
}
"
  success={[Function]}
  variables={
    {
      "datasetId": "gnomad_r3_controls_and_biobanks",
      "referenceGenome": "GRCh38",
      "transcriptId": "dummy-transcript",
    }
  }
>
  [Function]
</mockConstructor>
`;

exports[`MitochondrialTranscriptCoverageTrack with dataset gnomad_r3_non_cancer queries with proper parameters 1`] = `
<mockConstructor
  errorMessage="Unable to load coverage"
  loadingMessage="Loading coverage"
  loadingPlaceholderHeight={220}
  operationName="MitochondrialCoverageInTranscript"
  query="
query MitochondrialCoverageInTranscript($transcriptId: String!, $datasetId: DatasetId!, $referenceGenome: ReferenceGenomeId!) {
  transcript(transcript_id: $transcriptId, reference_genome: $referenceGenome) {
    mitochondrial_coverage(dataset: $datasetId) {
      pos
      mean
      median
      over_100
      over_1000
    }
  }
}
"
  success={[Function]}
  variables={
    {
      "datasetId": "gnomad_r3_non_cancer",
      "referenceGenome": "GRCh38",
      "transcriptId": "dummy-transcript",
    }
  }
>
  [Function]
</mockConstructor>
`;

exports[`MitochondrialTranscriptCoverageTrack with dataset gnomad_r3_non_neuro queries with proper parameters 1`] = `
<mockConstructor
  errorMessage="Unable to load coverage"
  loadingMessage="Loading coverage"
  loadingPlaceholderHeight={220}
  operationName="MitochondrialCoverageInTranscript"
  query="
query MitochondrialCoverageInTranscript($transcriptId: String!, $datasetId: DatasetId!, $referenceGenome: ReferenceGenomeId!) {
  transcript(transcript_id: $transcriptId, reference_genome: $referenceGenome) {
    mitochondrial_coverage(dataset: $datasetId) {
      pos
      mean
      median
      over_100
      over_1000
    }
  }
}
"
  success={[Function]}
  variables={
    {
      "datasetId": "gnomad_r3_non_neuro",
      "referenceGenome": "GRCh38",
      "transcriptId": "dummy-transcript",
    }
  }
>
  [Function]
</mockConstructor>
`;

exports[`MitochondrialTranscriptCoverageTrack with dataset gnomad_r3_non_topmed queries with proper parameters 1`] = `
<mockConstructor
  errorMessage="Unable to load coverage"
  loadingMessage="Loading coverage"
  loadingPlaceholderHeight={220}
  operationName="MitochondrialCoverageInTranscript"
  query="
query MitochondrialCoverageInTranscript($transcriptId: String!, $datasetId: DatasetId!, $referenceGenome: ReferenceGenomeId!) {
  transcript(transcript_id: $transcriptId, reference_genome: $referenceGenome) {
    mitochondrial_coverage(dataset: $datasetId) {
      pos
      mean
      median
      over_100
      over_1000
    }
  }
}
"
  success={[Function]}
  variables={
    {
      "datasetId": "gnomad_r3_non_topmed",
      "referenceGenome": "GRCh38",
      "transcriptId": "dummy-transcript",
    }
  }
>
  [Function]
</mockConstructor>
`;

exports[`MitochondrialTranscriptCoverageTrack with dataset gnomad_r3_non_v2 queries with proper parameters 1`] = `
<mockConstructor
  errorMessage="Unable to load coverage"
  loadingMessage="Loading coverage"
  loadingPlaceholderHeight={220}
  operationName="MitochondrialCoverageInTranscript"
  query="
query MitochondrialCoverageInTranscript($transcriptId: String!, $datasetId: DatasetId!, $referenceGenome: ReferenceGenomeId!) {
  transcript(transcript_id: $transcriptId, reference_genome: $referenceGenome) {
    mitochondrial_coverage(dataset: $datasetId) {
      pos
      mean
      median
      over_100
      over_1000
    }
  }
}
"
  success={[Function]}
  variables={
    {
      "datasetId": "gnomad_r3_non_v2",
      "referenceGenome": "GRCh38",
      "transcriptId": "dummy-transcript",
    }
  }
>
  [Function]
</mockConstructor>
`;

exports[`MitochondrialTranscriptCoverageTrack with dataset gnomad_sv_r2_1 queries with proper parameters 1`] = `
<mockConstructor
  errorMessage="Unable to load coverage"
  loadingMessage="Loading coverage"
  loadingPlaceholderHeight={220}
  operationName="MitochondrialCoverageInTranscript"
  query="
query MitochondrialCoverageInTranscript($transcriptId: String!, $datasetId: DatasetId!, $referenceGenome: ReferenceGenomeId!) {
  transcript(transcript_id: $transcriptId, reference_genome: $referenceGenome) {
    mitochondrial_coverage(dataset: $datasetId) {
      pos
      mean
      median
      over_100
      over_1000
    }
  }
}
"
  success={[Function]}
  variables={
    {
      "datasetId": "gnomad_sv_r2_1",
      "referenceGenome": "GRCh37",
      "transcriptId": "dummy-transcript",
    }
  }
>
  [Function]
</mockConstructor>
`;

exports[`MitochondrialTranscriptCoverageTrack with dataset gnomad_sv_r2_1_controls queries with proper parameters 1`] = `
<mockConstructor
  errorMessage="Unable to load coverage"
  loadingMessage="Loading coverage"
  loadingPlaceholderHeight={220}
  operationName="MitochondrialCoverageInTranscript"
  query="
query MitochondrialCoverageInTranscript($transcriptId: String!, $datasetId: DatasetId!, $referenceGenome: ReferenceGenomeId!) {
  transcript(transcript_id: $transcriptId, reference_genome: $referenceGenome) {
    mitochondrial_coverage(dataset: $datasetId) {
      pos
      mean
      median
      over_100
      over_1000
    }
  }
}
"
  success={[Function]}
  variables={
    {
      "datasetId": "gnomad_sv_r2_1_controls",
      "referenceGenome": "GRCh37",
      "transcriptId": "dummy-transcript",
    }
  }
>
  [Function]
</mockConstructor>
`;

exports[`MitochondrialTranscriptCoverageTrack with dataset gnomad_sv_r2_1_non_neuro queries with proper parameters 1`] = `
<mockConstructor
  errorMessage="Unable to load coverage"
  loadingMessage="Loading coverage"
  loadingPlaceholderHeight={220}
  operationName="MitochondrialCoverageInTranscript"
  query="
query MitochondrialCoverageInTranscript($transcriptId: String!, $datasetId: DatasetId!, $referenceGenome: ReferenceGenomeId!) {
  transcript(transcript_id: $transcriptId, reference_genome: $referenceGenome) {
    mitochondrial_coverage(dataset: $datasetId) {
      pos
      mean
      median
      over_100
      over_1000
    }
  }
}
"
  success={[Function]}
  variables={
    {
      "datasetId": "gnomad_sv_r2_1_non_neuro",
      "referenceGenome": "GRCh37",
      "transcriptId": "dummy-transcript",
    }
  }
>
  [Function]
</mockConstructor>
`;
